lib/elixir/src/elixir_fn.erl

```diff
@@ -178,18 +178,21 @@ format_error(clauses_with_different_arities) ->
 format_error(defaults_in_args) ->
   "anonymous functions cannot have optional arguments";
 format_error({block_expr_in_capture, Expr}) ->
-  io_lib:format("invalid args for &, block expressions are not allowed, got: ~ts",
+  io_lib:format("block expressions are not allowed inside the capture operator &, got: ~ts",
                 ['Elixir.Macro':to_string(Expr)]);
 format_error({nested_capture, Arg}) ->
-  io_lib:format("nested captures via & are not allowed: ~ts", ['Elixir.Macro':to_string(Arg)]);
+  io_lib:format("nested captures are not allowed. You cannot define a function using "
+    " the capture operator & inside another function defined via &. Got invalid nested "
+    "capture: ~ts", ['Elixir.Macro':to_string(Arg)]);
 format_error({invalid_arity_for_capture, Arity}) ->
-  io_lib:format("invalid arity for &, expected a number between 0 and 255, got: ~b", [Arity]);
+  io_lib:format("capture argument &~B must be a number between 0 and 255", [Arity]);
```

Comment:
Minor nitpick:

```suggestion
  io_lib:format("capture argument &~B is invalid (it must be a number between 0 and 255)", [Arity]);
```

Reply:
I prefer the current approach as it is more direct. The fact it is an error already reveals it is invalid. WDYT? :)

Reply:
What about?

```suggestion
  io_lib:format("capture argument &~B must be numbered between 0 and 255", [Arity]);
```